### ISSUE TITLE
Major Changes to JDK17

### ISSUE THREAD

**Issue by zachcp** [CDK-R/cdkr#126]
I received  the following from Brian Ripley:

```
Java 17 was released this week: as it is the first version officially
supporting arm64 Macs it is being used for the 'M1mac' additional
issues.  Packages

Myrrix RJDemetra RMOA expands jdx jsr223 rcdk rjdmarkdown tabulizer

fail, and this has been confirmed under Linux.  Pre-built binary
packages are already available from https://jdk.java.net/17/ and will
shortly be available from https://adoptium.net/.  As 17 is an LTS
release it is expected to become widely used over the next few months.

On the most common form of failure (all but jsr223, also seen in rJava),
Simon Urbanek writes

"There was a major change in JDK-17 that now disallows reflection to use
non-public access.  In short rJava can no longer pretend it is part of a
class when accessing methods/fields which allowed us to call methods
without explicitly subclassing a class.  This may have serious
repercussions, because code that could be before written in R now has to
be written in Java, compiled and then called from R."

Please correct before 2021-10-17 to safely retain your package on CRAN.
```

I can try to look into this over the weekend. Much of the rCDK is compiled so I don't think the affected surface area will be too bad but I expect the use of `$` for convenient reflection will be severely curtailed.

zach cp

**Comment by zachcp** [comment]
Additional Info from [the rJava Changelog](https://github.com/s-u/rJava/commit/f28f080a888d616f9a64750ea36f4b98dde62d40#diff-7ee66c4f1536ac84dc5bbff1b8312e2eef24b974b3e48a5c5c2bcfdf2eb8f3ceR4-R15).   

```
1.0-5	(under development)
    o	allow access modifications to fail on JDK-17

	== Important note to Java developers:
	JDK-17 no longer allows access overrides, so J() wrapper
	will fail for cases where the underlying class does
	not allow public access. This means that it is no longer
	possible to write high-level code that behaves as if it was a
	member of the class. Do not upgrade to JDK-17 if that is
	important in your application or write a Java wrapper that
	exposes the necessary fields/methods as public.
```

**Comment by zachcp** [comment]
I did a bit of work on this and ran into the classic "setting up a full R+Java dev environment" and got stuck.  However, since provisioning in CI is straightforward I setup up GH actions to [use JDK17 and run the tests](https://github.com/zachcp/cdkr/actions) (from [this branch](https://github.com/zachcp/cdkr/tree/j17)).  I tried to avoid the use of reflection by getting rid of `$` but we're still triggering errors...  so there may be deeper issues in the code.

 @rajarshi  there are only 2 weeks left to submit to updated packages to CRAN without risking package removal.  Any chance you can also take a crack at this?

**Comment by zachcp** [comment]
Addressed in #127

**Comment by zachcp** [comment]
addressed in #127

**Comment by zachcp** [comment]
Package on the way to CRAN. @rajarshi, once you merge #127 can you tag a release for 3.6.0?
